src/Debug/skeletonViewer.ts

```diff
@@ -388,20 +395,20 @@ export class SkeletonViewer {
         options.displayOptions.sphereScaleUnit = options.displayOptions.sphereScaleUnit ?? 2;
         options.displayOptions.sphereFactor = options.displayOptions.sphereFactor ?? 0.865;
         options.computeBonesUsingShaders = options.computeBonesUsingShaders ?? true;
-        options.useAllBones = options.useAllBones ?? true;        
-                
+        options.useAllBones = options.useAllBones ?? true;
+
         const initialMeshBoneIndices = mesh.getVerticesData(VertexBuffer.MatricesIndicesKind);
-        const initialMeshBoneWeights = mesh.getVerticesData(VertexBuffer.MatricesWeightsKind);        
-        const baseNodeBoneIndices = []
-        this._boneIndices = new Set(); 
-
-        if (options.useAllBones) {   
-            for (let i = 0; i < skeleton.bones.length; ++i){
-                baseNodeBoneIndices.push(skeleton.bones[i].getIndex(), 0, 0, 0)
-            }                    
+        const initialMeshBoneWeights = mesh.getVerticesData(VertexBuffer.MatricesWeightsKind);
+        const baseNodeBoneIndices = [];
+        this._boneIndices = new Set();
+
+        if (options.useAllBones) {
+            for (let i = 0; i < skeleton.bones.length; ++i) {
+                baseNodeBoneIndices.push(skeleton.bones[i].getIndex(), 0, 0, 0);
+            }
             for (let i = 0; i < baseNodeBoneIndices.length; ++i) {
-                this._boneIndices.add(baseNodeBoneIndices[i]);   
-            }            
+                this._boneIndices.add(baseNodeBoneIndices[i]);
+            }
         }else {

```

Comment:
I don't understand why you use `baseNodeBoneIndices`?

Reply:
I honestly dont know what you were doing with the this._boneIndices but I have to fill it in order for the bones to show up.

I was messing with some stuff at one point with how the indices were bound this is just renominates of it, I can pull it out I kinad forgot it was there.

I think it was because I did not like the using the Set stuff initially

Reply:
Ohh and thank you for the code overlook though!  I appreciate you.

Reply:
`_boneIndices` is simply the list of the indices of the bones used by the mesh. When displaying the skeleton, only bones with an index inside this set is displayed.

> I honestly dont know what you were doing with the this._boneIndices but I have to fill it in order for the bones to show up.

You can simply not check the bone index with `this._boneIndices` if `useAllBones = true`: see the code I given above.

Reply:
Ok cool, let me make those changes!